### PR TITLE
Some code cleanup

### PR DESCRIPTION
**Pull request by jaycarlton**
Various cleanup operations, mostly suggested by the IDE.

### CODE REVIEW

File: src/main/java/bio/terra/buffer/db/BufferDao.java [hunk 1]
```diff
@@ -144,6 +144,7 @@ public void createPools(List<Pool> pools) {
   }
 
   /** Updates list of pools' status to DEACTIVATED. */
+  // TODO: consider delaying expiration so pool can be recovered
```
Comment (jmczerk) [inline src/main/java/bio/terra/buffer/db/BufferDao.java:147]: I don't remember what our policy is about referencing Jira tickets in source code, but if we are doing that you might want to reference [PF-1415](https://broadworkbench.atlassian.net/browse/PF-1415).

File: src/main/java/bio/terra/buffer/service/resource/FlightSubmissionFactoryImpl.java [hunk 2]
```diff
@@ -38,7 +38,7 @@ public FlightSubmission getCreationFlightSubmission(Pool pool, ResourceId resour
 
   @Override
   public FlightSubmission getDeletionFlightSubmission(Resource resource, ResourceType type) {
-    if (!CREATION_FLIGHT_MAP.containsKey(type)) {
```
Comment (jmczerk) [inline src/main/java/bio/terra/buffer/service/resource/FlightSubmissionFactoryImpl.java:41]: Nice catch!

File: src/main/java/bio/terra/buffer/service/pool/PoolService.java [hunk 3]
```diff
@@ -161,68 +166,83 @@ public void updateFromConfig(List<PoolWithResourceConfig> parsedPoolConfigs) {
...
-                      .orElseThrow(
-                          () ->
-                              new RuntimeException(
-                                  String.format(
-                                      "Unknown ResourceType for PoolConfig %s", poolConfig))))
-              .status(PoolStatus.ACTIVE)
-              .build();
-      pools.add(createdPool);
-    }
-    bufferDao.createPools(pools);
```
Comment (jmczerk) [inline src/main/java/bio/terra/buffer/service/pool/PoolService.java:216]: Really like the refactor here, must ask the obligatory "does `createPools` have unit test coverage?" question...

**Review by jmczerk — APPROVED** [review]
The Codacy errors all looks like unused imports, otherwise LGTM.  Thanks for doing this, it improves readability a lot!

**Review by yuhuyoyo — APPROVED** [review]

File: src/main/java/bio/terra/buffer/db/BufferDao.java (hunk 1 above)
Comment (jaycarlton) [reply src/main/java/bio/terra/buffer/db/BufferDao.java:147]: Eh, it's not worth throwing away the passing tests and waiting  90 minutes.

File: src/main/java/bio/terra/buffer/service/pool/PoolService.java (hunk 3 above)
Comment (jaycarlton) [reply src/main/java/bio/terra/buffer/service/pool/PoolService.java:216]: It does as part of `updateFromConfig()`.

**Pull request merged**